sound/soc/bcm/allo-piano-dac-plus.c

```diff
@@ -399,14 +398,14 @@ static int pcm512x_get_reg_sub(struct snd_kcontrol *kcontrol,
 	unsigned int right_val = 0;
 
 	rtd = snd_soc_get_pcm_runtime(card, card->dai_link[0].name);
-	right_val = snd_soc_read(rtd->codec_dais[1]->codec,
-			PCM512x_DIGITAL_VOLUME_3);
+	snd_soc_component_read(rtd->codec_dais[1]->component,
+			PCM512x_DIGITAL_VOLUME_3, &right_val);
 	if (right_val < 0)
 		return right_val;
 
 	if (glb_ptr->dual_mode != 1) {
-		left_val = snd_soc_read(rtd->codec_dais[1]->codec,
-				PCM512x_DIGITAL_VOLUME_2);
+		snd_soc_component_read(rtd->codec_dais[1]->component,
+				PCM512x_DIGITAL_VOLUME_2, &left_val);
 		if (left_val < 0)
 			return left_val;
 
```

Comment:
this should be
```
ret = snd_soc_component_read(...);
if (ret)
    return ret;
```
And there's a bunch more of this (checking read value instead of return value) further down the file